packages/components/form/src/form-item.vue

```diff
@@ -192,16 +196,25 @@ const _rules = computed(() => {
   }
 
   if (props.required !== undefined) {
-    rules.push({ required: !!props.required })
+    // remove inconsistent rules
+    rules = rules.filter(
+      (rule) =>
+        rule.required === undefined || !!rule.required === !!props.required
```

Comment:
May cause rules to be lost.
```js
props.rule = [{max:9}]
```

Reply:
It won't. `rule.required === undefined` is `true`, so it can't be filtered.

Reply:
if `props.rule = [{ max: 9, require: true }]` and  `props.required = false`

Reply:
Updated.

Reply:
approved.